tests/core/v5_1/alexandria/test_alexandria_network.py

```diff
@@ -102,3 +105,126 @@ async def test_alexandria_network_responds_to_find_nodes(
 
     assert len(enrs) >= 1
     assert any(enr.node_id == enr.node_id for enr in enrs)
+
+
+@pytest.mark.trio
+async def test_alexandria_network_get_content_proof_api(
+    alice, bob, alice_alexandria_network, bob_alexandria_client,
+):
+    content = ContentFactory(length=16 * 32 * 4)
+    proof = compute_proof(content, sedes=content_sedes)
+
+    async with bob_alexandria_client.subscribe(GetContentMessage) as subscription:
```

Comment:
`GetContentMessage` handles for `get_content_from_nodes` and `get_content_proof`? What's the thinking behind not having a separate proof message type?

Reply:
`get_content_proof` is the mid-level primative for retrieving a proof for some content from a single node.

`get_content_from_nodes` is used to retrieve the full content.  It assumes you have figured out all the nodes that have the content you want.  Internally it uses `get_content_proof` to retrieve the content in pieces, re-assembling it locally, and returning the full content once it has been fully retrieved.

Reply:
Oh, as for the separation of not having something like `RawContent` and `ContentProof` message types...  It's just really simple to have a single boolean flag and use the same message type.

We also don't have infrastructure for a request that has two different response types.

And we also can't know ahead of time if the content we're requesting will fit inside a single message or need to be split across multiple messages in a proof...